source/tests/test.ts

```diff
@@ -1,147 +1,246 @@
+			async () => {} // tslint:disable-line:no-empty
+		]}
+	], ['map',
+		{is: m.map, fixtures: [
+			new Map()
+		]}
+	], ['set',
+		{is: m.set, fixtures: [
+			new Set()
+		]}
+	], ['weakSet',
+		{is: m.weakSet, fixtures: [
+			new WeakSet()
+		]}
+	], ['weakMap',
+		{is: m.weakMap, fixtures: [
+			new WeakMap()
+		]}
+	], ['int8Array',
+		{is: m.int8Array, fixtures: [
+			new Int8Array(0)
+		]}
+	], ['uint8Array',
+		{is: m.uint8Array, fixtures: [
+			new Uint8Array(0)
+		]}
+	], ['uint8ClampedArray',
+		{is: m.uint8ClampedArray, fixtures: [
+			new Uint8ClampedArray(0)
+		]}
+	], ['int16Array',
+		{is: m.int16Array, fixtures: [
+			new Int16Array(0)
+		]}
+	], ['uint16Array',
+		{is: m.uint16Array, fixtures: [
+			new Uint16Array(0)
+		]}
+	], ['int32Array',
+		{is: m.int32Array, fixtures: [
+			new Int32Array(0)
+		]}
+	], ['uint32Array',
+		{is: m.uint32Array, fixtures: [
+			new Uint32Array(0)
+		]}
+	], ['float32Array',
+		{is: m.float32Array, fixtures: [
+			new Float32Array(0)
+		]}
+	], ['float64Array',
+		{is: m.float64Array, fixtures: [
+			new Float64Array(0)
+		]}
+	], ['arrayBuffer',
+		{is: m.arrayBuffer, fixtures: [
+			new ArrayBuffer(10)
+		]}
+	], ['nan',
+		{is: m.nan, fixtures: [
+			NaN,
+			Number.NaN
+		]}
+	], ['nullOrUndefined',
+		{is: m.nullOrUndefined, fixtures: [
+			null,
+			undefined
+		]}
+	], ['plainObject',
+		{is: m.plainObject, fixtures: [
+			{x: 1},
+			Object.create(null),
+			new Object()
+		]}
+	], ['integer',
+		{is: m.integer, fixtures: [
+			6
+		]}
+	], ['safeInteger',
+		{is: m.safeInteger, fixtures: [
+			Math.pow(2, 53) - 1,
+			-Math.pow(2, 53) + 1
+		]}
+	], ['domElement',
+		{is: m.domElement, fixtures: [
+			'div',
+			'input',
+			'span',
+			'img',
+			'canvas',
+			'script'
+		].map(createDomElement) }
+	], ['non-domElements',
+		{is: value => !m.domElement(value), fixtures: [
+			document.createTextNode('data'),
+			document.createProcessingInstruction('xml-stylesheet', 'href="mycss.css" type="text/css"'),
+			document.createComment('This is a comment'),
+			document,
+			document.implementation.createDocumentType('svg:svg', '-//W3C//DTD SVG 1.1//EN', 'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'), // tslint:disable-line
+			document.createDocumentFragment()
+		]}
+	], ['infinite',
+		{is:  m.infinite, fixtures: [
+			Infinity,
+			-Infinity
+		]}
+	]
 ]);
 
-// This ensures a certain method matches only the types
-// it's supposed to and none of the other methods' types
-const testType = (t, type, exclude) => {
-	for (const [key, value] of types) {
+/* This ensures a certain method matches only the types
+	 it's supposed to and none of the other methods' types */
+const testType = (t: TestContext & Context<any>, type: string, exclude?: string[]) => {
+	const testData = types.get(type);
+
+	if (testData === undefined) {
+		t.fail(`is.${type} not defined`);
+
+		return;
+	}
+
+	const {is} = testData;
+
+	types.forEach(({fixtures}, key) => {
```

Comment:
Use a `for..of` instead (like it was).

```ts
for (const [key, {fixtures}] of types) {
	// TODO: Automatically exclude value types in other tests that we have in the current one.
	// Could reduce the use of `exclude`.
	if (exclude && exclude.indexOf(key) !== -1) {
		continue;
	}

	const assert = key === type ? t.true.bind(t) : t.false.bind(t);

	for (const fixture of fixtures) {
		assert(is(fixture), `Value: ${util.inspect(fixture)}`);
	}
}
```

Reply:
Also, there's the comment about the exclude, we can use `types.filter` to get rid of it

Reply:
Also, there's the comment about the exclude, we can use `types.filter` to get rid of it

Reply:
not unless we do `Array.from(types).filter()` since `types` is a Map, but feel free to do so, If you feel that that would be an improvement :v: